mino/minogrpc/server.go

```diff
@@ -472,17 +549,26 @@ func (r receiver) Recv(ctx context.Context) (mino.Address, proto.Message, error)
 
 	enveloppe := &Envelope{}
 	err = ptypes.UnmarshalAny(msg.Message, enveloppe)
-	if err != nil {
-		return nil, nil, encoding.NewAnyDecodingError(msg.Message, err)
+	// we can receive two types of messages: enveloppes and other messages. If
```

Comment:
Why don't you make the logic easier by always sending an envelope/overlay message ? When a message is received, you check that the address is yours or if you should relay.